server/game/cards/characters/02/nymeriasand.js

```diff
@@ -1,69 +1,35 @@
-const _ = require('underscore');
-
 const DrawCard = require('../../../drawcard.js');
 
 class NymeriaSand extends DrawCard {
     setupCardAbilities(ability) {
         this.action({
-            title: 'Remove icon from opponent\'s character',
-            method: 'stealIcon',
-            limit: ability.limit.perPhase(1)
-        });
-    }
-
-    stealIcon(player) {
-        if(this.location !== 'play area' || player.phase !== 'challenge') {
-            return false;
-        }
-
-        this.game.promptForSelect(player, {
-            cardCondition: card => this.cardCondition(card),
-            activePromptTitle: 'Select a character to remove an icon from',
-            source: this,
-            onSelect: (player, card) => this.onCardSelected(player, card)
-        });
-
-        return true;
-    }
-
-    cardCondition(card) {
-        return card.getType() === 'character' && card.controller !== this.controller;
-    }
-
-    onCardSelected(player, card) {
-        var icons = ['Military', 'Intrigue', 'Power'];
-
-        this.selectedCard = card;
-
-        var buttons = _.map(icons, icon => {
-            return { text: icon, method: 'iconSelected', arg: icon.toLowerCase() };
-        });
-
-        this.game.promptWithMenu(this.controller, this, {
-            activePrompt: {
-                menuTitle: 'Select an icon to remove',
-                buttons: buttons
+            title: 'Remove and gain icon',
+            limit: ability.limit.perPhase(1),
+            phase: 'challenge',
+            target: {
+                activePromptTitle: 'Select a character',
+                cardCondition: card => card.location === 'play area' && card.controller !== this.controller &&
+                                       card.getType() === 'character'
             },
-            source: this
+            handler: context => {
+                this.game.promptForIcon(this.controller, context);
```

Comment:
I'm not sure I like how this is used. While putting it onto the context object isn't a bad idea, the fact that it's async but doesn't appear async is a problem. It would be very easy to forget to do the `queueSimpleStep`s you've added.

Couple options: we could make this a flag on abilities, so that you're not calling it directly. By the time it reaches the handler, it will have the icon on the context object like you've done.
```javascript
this.action({
    // ...
    chooseIcon: true,
    handler: context => {
        // do something with context.icon
    }
});
```

Or, to make the async nature more clear, the prompt should take a callback instead of the context object.
```javascript
this.game.promptForIcon(this.controller, icon => {
    // do something with icon
});
```